LibraBFT/Impl/Consensus/RoundManager.agda

```diff
@@ -146,20 +149,18 @@ module ProcessProposalM (proposal : Block) where
          -- > executeAndVoteM proposal >>= \case
          -- is translated to the following.
            executeAndVoteM proposal >>= step₂
-  step₂ r =
-         case r of λ where
-           (Left _) → logErr -- <propagate error>
-           (Right vote) → step₃ vote
-  step₃ vote = do
-             RoundState.recordVote vote
-             si ← BlockStore.syncInfoM
-             step₄ vote si
-  step₄ vote si = do
-             recipient ← ProposerElection.getValidProposer
-                         <$> use lProposerElection
-                         <*> pure (proposal ^∙ bRound + 1)
-             act (SendVote (VoteMsg∙new vote si) (recipient ∷ []))
-             -- TODO-2:                                                mkNodesInOrder1 recipient
+  step₂ =  λ where
+            (Left _) → logErr -- <propagate error>
+            (Right vote) → do
+              RoundState.recordVote vote
+              si ← BlockStore.syncInfoM
+              step₃ vote si
+  step₃ vote si = do
```

Comment:
Why is step3 needed?
It seems there are no conditionals nor state changes until the final `act`.

Reply:
The `steps` are useful for code that *follows* some call to a complex operation, not code that contains it.

Reply:
But `BlockStore.syncInfoM` is just a convenience accessor - it just grabs three fields and wraps them (done in the LBFT context).
Curious, how is that complex in terms of the proof?

Reply:
Ah, well looking more closely, I see it's this way because `syncInfoM` is postulated. I can implement it, then this step can be easily removed.